plugins/components/controllable/Controllable.hpp

```diff
@@ -156,8 +162,8 @@ namespace Components {
         void deserialize(std::vector<uint8_t> &data) override {
             if (data.size() != sizeof(__data))
                 throw std::runtime_error("Invalid data size for Controllable component");
-            __network.inputs = *reinterpret_cast<std::array<bool, 4> *>(data.data());
-            __network.actions = *reinterpret_cast<std::array<bool, 10> *>(data.data() + 4);
+            inputs = *reinterpret_cast<std::array<bool, 4> *>(data.data());
+            actions = *reinterpret_cast<std::array<bool, 10> *>(data.data() + 4);
```

Comment:
I know you didn't just write what I'm about to comment about, but what on earth is this `reinterpret_cast` from a raw pointer to a pointer to an `std::array`??

In practice we will certainly not notice this since we are not going to deserialise this component, but IMO this will definitely crash as soon as we call it

Reply:
Changed it